pkg/registry/generic/rest/streamer.go

```diff
@@ -25,13 +25,19 @@ import (
 	"k8s.io/kubernetes/pkg/api/rest"
 )
 
+// A response checker to transform the error.
+type LocationResponseChecker interface {
+	Check(resp *http.Response) error
```

Comment:
Add comment on expectations for implementations-- e.g. under what conditions should/must they close resp.Body?